plugins/Live/Live.php

```diff
@@ -57,69 +52,99 @@ public function addJsGlobalVariables(&$out)
+    /**
+     * Throws an exception if visits log is disabled
+     *
+     * @param null|int|array $idSite
+     * @throws \Exception
+     */
+    public static function checkIsVisitorLogEnabled($idSite = null): void
     {
-        [$profileEnabled, $logEnabled] = self::getSettings($idSite);
+        $systemSettings = new SystemSettings();
 
-        return $profileEnabled;
-    }
+        if ($systemSettings->activateVisitorLog->getValue() == false) {
+            throw new \Exception('Visits log is deactivated globally');
+        }
 
-    private static function getSettings($idSite = null)
-    {
         if (empty($idSite)) {
             $idSite = Common::getRequestVar('idSite', 0, 'int');
         }
 
-        $cache = Cache::getTransientCache();
-        $siteIdLoaded = $cache->fetch(self::CurrentSiteCacheKey);
-        $visitorProfileCached = $cache->contains(self::ProfileEnabledCacheKey);
-        $visitorLogCached = $cache->contains(self::LogEnabledCacheKey);
+        if (!empty($idSite)) {
+            $idSites = is_array($idSite) ? $idSite : [$idSite];
 
-        if ($visitorProfileCached && $visitorLogCached && $idSite == $siteIdLoaded) {
-            return [
-                $cache->fetch(self::ProfileEnabledCacheKey),
-                $cache->fetch(self::LogEnabledCacheKey),
-            ];
-        }
+            foreach ($idSites as $idSite) {
+                $settings = new MeasurableSettings($idSite);
 
-        $siteIdLoaded = $idSite;
-        $visitorProfileEnabled = true;
-        $visitorLogEnabled = true;
+                if ($settings->activateVisitorLog->getValue() == false) {
+                    throw new \Exception('Visits log is deactivated in website settings');
+                }
+            }
+        }
+    }
 
+    /**
+     * Returns whether visits log is enabled (for the given site)
+     *
+     * @param null|int|array $idSite
+     * @return bool
+     */
+    public static function isVisitorLogEnabled($idSite = null): bool
+    {
         try {
-            if (!empty($idSite)) {
-                $settings = new MeasurableSettings($idSite);
+            self::checkIsVisitorLogEnabled($idSite);
+        } catch (\Exception $e) {
+            return false;
+        }
 
-                $visitorProfileEnabled = $settings->activateVisitorProfile->getValue();
-                $visitorLogEnabled     = $settings->activateVisitorLog->getValue();
-            }
+        return true;
+    }
+    /**
+     * Throws an exception if visitor profile is disabled
+     *
+     * @param null|int|array $idSite
+     * @throws \Exception
+     */
+    public static function checkIsVisitorProfileEnabled($idSite = null): void
+    {
+        $systemSettings = new SystemSettings();
 
-            $systemSettings = new SystemSettings();
+        if ($systemSettings->activateVisitorProfile->getValue() == false) {
+            throw new \Exception('Visitor profile is deactivated globally');
```

Comment:
We could maybe add something like `. A user with super user access can enable this feature in the general settings.`?